src/main/java/com/chrosciu/bootcamp/tasks/github/GithubClient.java

```diff
@@ -10,22 +10,30 @@ public class GithubClient {
     private final GithubApi githubApi;
 
     public Flux<Repository> getUserRepositories(String username) {
-        //TODO: Implement
-        return null;
+        return githubApi.getUserRepositories(username)
+                .flatMapMany(Flux::fromIterable);
     }
 
     public Flux<Branch> getUserRepositoryBranches(String username, String repo) {
-        //TODO: Implement
-        return null;
+        return githubApi
+                .getUserRepositoryBranches(username, repo)
+                .flatMapMany(Flux::fromIterable);
     }
 
     public Flux<Repository> getUsersRepositories(Flux<String> usernames) {
-        //TODO: Implement
-        return null;
+        Flux<Repository> flux = usernames
+                .map(githubApi::getUserRepositories)
+                .flatMap(elem -> {
+                    return elem.flatMapMany(Flux::fromIterable);
+                });
+        return flux;
     }
 
     public Flux<String> getAllUserBranchesNames(String username) {
-        //TODO: Implement
-        return null;
+        return githubApi.getUserRepositories(username)
+                .flatMapMany(Flux::fromIterable)
+                .flatMap(repo -> githubApi.getUserRepositoryBranches(username, repo.getName()))
+                .flatMap(Flux::fromIterable)
+                .map(o -> o.toString());
```

Comment:
to zwróci całą reprezentację obiektu - coś w stylu `Branch[name=]` a nie samo name :)